src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -319,34 +342,56 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   // For Control frames Data Size value shall be less than MTU header
   // For Single and Consecutive Data Size value shall be greater than 0x00
   // and shall be less than payload size
-  if (header.dataSize > payload_size) {
-    LOG4CXX_WARN(logger_,
-                 "Packet data size is "
-                     << header.dataSize
-                     << " and bigger than allowed payload size " << payload_size
-                     << " bytes");
-    return RESULT_FAIL;
-  }
+  // For First Frame Data Size shall be equal 0x08 (payload of this packet will
+  // be 8 bytes).
+
   switch (header.frameType) {
     case FRAME_TYPE_SINGLE:
     case FRAME_TYPE_CONSECUTIVE:
-      if (header.dataSize <= 0u) {
+    case FRAME_TYPE_CONTROL: {
+      bool wrongDataSize = false;
+
+      if (header.dataSize > payload_size) {
+        wrongDataSize = true;
+      }
+
+      if (FRAME_TYPE_CONTROL != header.frameType && header.dataSize <= 0u) {
+        wrongDataSize = true;
+      }
+
+      if (wrongDataSize) {
+        const std::string str_frame_type = StringifyFrameType(header.frameType);
```

Comment:
there is no need to create a variable for values used only one time. please just put the method call in the log call

Reply:
@iCollin deleted variable but I add `UNUSED()` for fix build with `Enable_log - off ` [0638eaf](https://github.com/smartdevicelink/sdl_core/pull/2516/commits/0638eafbedee6ca6f9401e6f8aaa2514839507b2)